packages/app/ui/buttons/likeButton.tsx

```diff
@@ -0,0 +1,44 @@
+import { Pressable } from "app/design-system";
+import { tw } from "app/design-system/tailwind";
+import Like from "app/ui/icons/like";
+import {
+  Entry,
+  useLikeEntryMutation,
+  useUserLikesQuery,
+} from "app/api/graphql";
+import useLikeCache from "app/hooks/useLikeCache";
+import { toast } from "app/utils/toast";
+import { ErrorType } from "app/types";
+
+type Props = {
+  size: number;
+  className?: string;
+  entry: Entry;
+};
+
+export function LikeButton({ size, className, entry }: Props) {
+  const [likeEntry] = useLikeEntryMutation();
+  const { data: userLikesData } = useUserLikesQuery();
+  const { addLikeToCache, removeLikeFromCache } = useLikeCache();
+  const active = !!userLikesData?.userLikes?.find((e) => e?.id === entry.id);
+
+  const update = async () => {
+    if (!entry.id) return;
+    active ? removeLikeFromCache(entry) : addLikeToCache(entry);
+    try {
+      await likeEntry({ variables: { id: entry.id, like: !active } });
+    } catch (e) {
+      active ? addLikeToCache(entry) : removeLikeFromCache(entry);
```

Comment:
```suggestion
      active ?  removeLikeFromCache(entry) : addLikeToCache(entry);
```

Reply:
Why?

Reply:
but active is captured before state was changed from the view of the function it is still true, or am iI wrong there?

Reply:
Try to turn wifi off and unlike/like something, it should be working as intended

Reply:
hmm, okay then.